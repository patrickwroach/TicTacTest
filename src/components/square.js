import React, {Component} from 'react';

export class Square extends React.Component {
  
  render() {
    return (
      <div className={"square " + this.props.status} onClick={() => this.props.onClick()}>
          <div class="status-container">
          <div class="x-shape">
            <div class="slash"></div>
            <div class="slash"></div>
          </div>
          <div class="o-shape"></div>
        </div>        
      </div>
    );
  }
}
