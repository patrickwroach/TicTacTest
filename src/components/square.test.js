import React from 'react';
import { shallow } from 'enzyme';
import {Square} from './square';

describe('Square', () => {
     
    const grid = {
        squares: Array(9).fill(null)
      };
      const squareClick = jest.fn();  

  it('It should render correctly in "debug" mode', () => {  

    const component = shallow(<Square 
        debug grid={grid}
         />);  
    expect(component).toMatchSnapshot();
  });

  it('Squares should fire a function on click', () => {
    const component = shallow(<Square debug grid={grid.squares} onClick={squareClick}
       />);  
       component
       .find('.square')
       .simulate('click');
       expect(squareClick).toHaveBeenCalled();
  });
});






