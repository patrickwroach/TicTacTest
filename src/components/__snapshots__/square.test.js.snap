// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square It should render correctly in "debug" mode 1`] = `
<div
  className="square undefined"
  onClick={[Function]}
>
  <div
    class="status-container"
  >
    <div
      class="x-shape"
    >
      <div
        class="slash"
      />
      <div
        class="slash"
      />
    </div>
    <div
      class="o-shape"
    />
  </div>
</div>
`;
